File: lesailes_api.py
import requests

def get_categories():
    url = "https://api.lesailes.uz/api/categories"
    
    try:
        response = requests.get(url)
        response.raise_for_status()  # Raise an exception for bad status codes
        
        data = response.json()
        return data
    except requests.exceptions.RequestException as e:
        print(f"Error fetching data: {e}")
        return None

if __name__ == "__main__":
    categories = get_categories()
    if categories:
        print("Categories fetched successfully!")
        print(categories) 

File: lesailes_api.js
async function getCategories() {
    try {
        const response = await fetch('https://api.lesailes.uz/api/categories/root');
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('Error fetching categories:', error);
        return null;
    }
}

// Run the function
getCategories()
    .then(data => {
        if (data) {
            console.log('Categories fetched successfully!');
            console.log(JSON.stringify(data, null, 2));
        }
    })
    .catch(error => {
        console.error('Error:', error);
    }); 